### ISSUE TITLE
Gather-scatter LU decomposition for parallel RBF mappings

### ISSUE THREAD

**Issue by uekerman** [precice/precice#718]
**The Problem**

preCICE currently offers two ways how to solve an RBF mapping system:
* Using GMRES in every call to `map`. To this end, we use PETSc, which also works in parallel.
https://github.com/precice/precice/blob/develop/src/mapping/PetRadialBasisFctMapping.hpp
* Computing an LU decomposition in `computeMapping` and solving one forward and one backward substitution in every `map`. Here, we use Eigen. However, our implementation only works for a serial participant.
https://github.com/precice/precice/blob/develop/src/mapping/RadialBasisFctMapping.hpp

PETSc keeps being a tedious build dependency. Already now, it is optional, but then RBF mappings cannot be used in parallel. 
Furthermore, GMRES in every timestep is not necessarily the best option. Depending on the mesh size and the number of timesteps, LU might also be a valid alternative performance-wise.

**Suggested Solution**

Extend the LU-based Eigen implementation to also handle a parallel participant. This should be done, as a first variant, in a brute-force gather-scatter way:
* Gather both meshes during `computeMapping` and compute LU decomposition on master.
* Gather data during `map`, solve forward and backward substitution on master, scatter data.

Communication should be handled within `RadialBasisFctMapping`, also all global data structures. Meaning, the mesh on the master rank should not contain the global mesh.

Then, we could switch off PETSc as a build dependency by default and in all binaries and users with large cases could still switch it on if needed. `mapping:rbf` should use GMRES / PETSc if built with PETSc, LU / Eigen otherwise or if configured explicitly (so very similarly to how it is handled right now already): https://github.com/precice/precice/blob/develop/src/mapping/config/MappingConfiguration.cpp#L272-L376   

This will be a very slow and inefficient variant if larger meshes are used. This is however not the use case, so we could ignore this at first. Later, we could, however, also look into a proper parallel LU decomposition (such as discussed in the Parallel Numerics Lecture at TUM) and implement it with Eigen (no new dependency if possible!). But this could be formulated in another issue.
